funsor/tensor.py

```diff
@@ -151,19 +151,21 @@ def __init__(self, data, inputs=None, dtype="real"):
 
     @ignore_jit_warnings()
     def __repr__(self):
-        if self.output != "real":
-            return "Tensor({}, {}, {})".format(self.data, self.inputs, repr(self.dtype))
+        inputs = dict.__repr__(self.inputs)
+        if self.dtype != "real":
+            return "Tensor({}, {}, {})".format(self.data, inputs, repr(self.dtype))
         elif self.inputs:
-            return "Tensor({}, {})".format(self.data, self.inputs)
+            return "Tensor({}, {})".format(self.data, inputs)
```

Comment:
WDYT of stripping `tensor(...)` here? So we get `Tensor([1, 2, 3])` instead of `Tensor(tensor([1, 2, 3]))`.

Reply:
Hmm I'd prefer for now to be more verbose so that the result can be evaluated back to the original.

In the future I think we could elide the `tensor(...)` if we made `TensorMeta` promote `list` objects via `ops.make_array()` or something (dispatching to `np.array` or `torch.tensor`), but that seems out of scope for this PR. Feel free to add an `ops.make_array()` and change this, though.

Reply:
Makes sense to me. Thanks for explaining.

Reply:
(by the way, asking for an opinion is not a request - so no need to put a thumb down)